### PR TITLE
[terra-dev-site] Bootstrap dev site entries to allow for chunking

### PR DESCRIPTION
**Pull request by mjhenkes**
### Summary

Webpack entries by default are independent chunks. Currently we have a bunch of includes and data in the entry. This results in several sizable chunks, one for each entry.

Through investigating module federation I learned that by loading a 'bootstrap' file containing all the information that was normally in the chunk as a side effect, it allows webpack to optimize the contents of the entrypoint and collect shared dependencies in a common bundle. This allows us to reduce the total bundle size by a ~1 MB for terra-dev-site. Also load times for the second site would be reduced due to caching of the shared bundle.

Using this snapshot in terra-ui where we only have one site increased the bundle size by ~1kb which seems like a decent trade off. Build times do not seem to be greatly affected by adding the additional loader.

Before:

Screen Shot 2021-09-15 at 3 11 23 PM

After: 

Screen Shot 2021-09-15 at 3 12 38 PM

Closes #

### Deployment Link

https://terra-applic-.herokuapp.com/

### Additional Details

Thank you for contributing to Terra.
@cerner/terra

### CODE REVIEW

File: packages/terra-dev-site/src/webpack/loaders/devSiteBootstrap.js [hunk 1]
```diff
@@ -0,0 +1,83 @@
+const lodashTemplate = require('lodash.template');
```
Comment (mjhenkes) [inline packages/terra-dev-site/src/webpack/loaders/devSiteBootstrap.js:1]: this loader was renamed from devSiteEntry.js to devSiteBootstrap.js

File: packages/terra-dev-site/src/webpack/loaders/devSiteBootstrap.js [hunk 2]
```diff
@@ -0,0 +1,83 @@
...
+
+  const {
+    entryPath,
+    siteConfig,
+    basename,
+    resolveExtensions,
+    sites,
+    contentDirectory,
+    isLernaMonoRepo,
+  } = getOptions(this).bootstrapLoaderData[this.resourceQuery];
```
Comment (mjhenkes) [inline packages/terra-dev-site/src/webpack/loaders/devSiteBootstrap.js:36]: bootstrapLoaderData was renamed from entryLoaderData

File: packages/terra-dev-site/src/webpack/templates/bootstrap.template [hunk 3]
```diff
@@ -0,0 +1,43 @@
+<% sideEffectImportFilePaths.forEach(sideEffect => print(`import '${sideEffect}';\n`)); %>
```
Comment (mjhenkes) [inline packages/terra-dev-site/src/webpack/templates/bootstrap.template:1]: This template was renamed from entry.template to bootstrap.template

no other changes were made.

File: packages/terra-dev-site/src/webpack/templates/entry.template [hunk 4]
```diff
@@ -1,43 +1 @@
...
-  enableDebugLogging: <%= enableDebugLogging %>,
-  additionalConfig: <%= additionalConfig %>,
-};
-
-if (siteConfig.enableDebugLogging) {
-  console.log('siteConfig', siteConfig);
-}
-
-ReactDOM.render(<Entry siteConfig={siteConfig} />, document.getElementById('root'));
+import('./bootstrap.template<%= key %>');
```
Comment (mjhenkes) [inline packages/terra-dev-site/src/webpack/templates/entry.template]:
```suggestion
import('./bootstrap.template<%= key %>');

```

File: packages/terra-dev-site/tests/jest/webpack/loaders/devSiteBootstrap.test.jsx [hunk 5]
```diff
@@ -0,0 +1,152 @@
+jest.mock('../../../../src/webpack/loaderUtils/generateNavigationConfig');
```
Comment (mjhenkes) [inline packages/terra-dev-site/tests/jest/webpack/loaders/devSiteBootstrap.test.jsx:1]: this test file was renamed from devSiteEntry.test.jsx to devSiteBootstrap.test.jsx

**Comment by mjhenkes** [comment]
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true">Fails</th>
    </tr>
  </thead>
  <tbody><tr>
      <td>:no_entry_sign:</td>
      <td>Please include a CHANGELOG entry for each changed package on this PR. Looks like a CHANGELOG is missing for: 

 - terra-dev-site</td>
    </tr>
  </tbody>
</table>




<p align="right">
  Generated by :no_entry_sign: <a href="https://danger.systems/js">dangerJS</a> against 4dbbecd9628531c3b11c79f48162e3bb0ccd7413
</p>

File: webpack.config.js [hunk 6]
```diff
@@ -8,11 +8,10 @@ const html = fs.readFileSync(require.resolve('./tests/terra-dev-site/head.html')
 
 const devSiteConfig = (env = {}, argv = { p: false }) => ({
   // temporary fix to enable live reloading.
-  target: argv.p || argv.mode === 'production' ? 'browserslist' : 'web',
+  target: argv.p || argv.mode === 'production' ? 'browserslist' : 'browserslist',
```
Comment (benbcai) [inline webpack.config.js]: So the targeted environment will always be `browserslist`?

Comment (mjhenkes) [reply webpack.config.js]: oh whoops, this wasn't supposed to be checked in. you need to have browserlist set for webpack dev server to work in ie, but there is a bug in webpack dev server that breaks hot reloading when you have browserlist set... it's great.

Comment (mjhenkes) [reply webpack.config.js]:
```suggestion
  target: argv.p || argv.mode === 'production' ? 'browserslist' : 'web',
```

**Review by benbcai — APPROVED** [review]

**Review by ryanthemanuel — APPROVED** [review]

**Pull request merged**